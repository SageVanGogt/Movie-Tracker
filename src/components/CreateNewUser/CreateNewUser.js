import React, { Component } from "react";

class CreateNewUser extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: "",
      email: "",
      password: ""
    };
  }

  handleChange = (event) => {
    const {name, value} = event.target
    this.setState({
        [name]: value
    })
  };

  handleSubmit = (event) => {
    event.preventDefault();

  };

  render() {
    return (
      <form type="submit" onChange={this.handleSubmit}>
        <input
          type="text"
          value={this.state.name}
          name="name"
          onChange={this.handleChange}
        />
        <input
          type="text"
          value={this.state.email}
          name="email"
          onChange={this.handleChange}
        />
        <input
          type="text"
          value={this.state.password}
          name="password"
          onChange={this.handleChange}
        />
      </form>
    );
  }
}

export default CreateNewUser