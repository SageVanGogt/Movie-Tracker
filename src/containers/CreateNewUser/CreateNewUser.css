.new-user-form {
  display: inline-block;
  margin: 20px;
  box-sizing: border-box;
  height: 250px;
}

.border {
  border-right: 2px soild #FFF; 
}

.form-field {
  display: block;
  width: 350px;
  height: 30px;
  margin: 15px;
  padding-left: 10px;
}

.submit-button {
  display: block;
  width: 350px;
  background-color: #a108ff;
  color: #FFF;
  height: 30px;
  margin: 15px;
  box-sizing: content-box;
  border: none;
  cursor: pointer;
}

input[type="submit"]:hover {
  background-color: #b748fc;
}

.login-header {
  color: #FFF;
  font-size: 1.3rem;
  font-weight: 400;
  margin: 15px;

}