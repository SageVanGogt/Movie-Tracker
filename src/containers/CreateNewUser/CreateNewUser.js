import React, { Component } from "react";
import { addUserFetch, validateEmail } from "../../apiCall/apiCall";
import { connect } from 'react-redux';
import { updateStoreUser, addError } from './../../actions/index';
import PropTypes from 'prop-types';
import './CreateNewUser.css';

export class CreateNewUser extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: "",
      email: "",
      password: ""
    };
  }

  handleChange = (event) => {
    const {name, value} = event.target;
    this.setState({
      [name]: value
    });
  };

  checkValidEmail = async (event) => {
    event.preventDefault();    
    const actual = await validateEmail(this.state.email);
    if (actual.isValid) {
      this.handleSubmit();
    } else {
      this.props.handleError('That email is not valid') ;
    }
  };

  handleSubmit = async () => {
    const response = await addUserFetch(this.state);
    if (response.id) {
      this.props.handleSignup({id: response.id, name: this.state.name});
    } else if (response.error) {
      this.props.handleError("Email already in use");
    }
  };

  render() {
    return (
      <form 
        type="submit"
        className="new-user-form border" 
        onSubmit={this.checkValidEmail}
      >
        <h2 className="login-header">SIGN UP</h2>
        <input
          className = "form-field"
          type="text"
          value={this.state.name}
          name="name"
          onChange={this.handleChange}
          placeholder="name"          
        />
        <input
          className="form-field"
          type="text"
          value={this.state.email}
          name="email"
          onChange={this.handleChange}
          placeholder="email"          
        />
        <input
          className="form-field"
          type="text"
          value={this.state.password}
          name="password"
          onChange={this.handleChange}
          placeholder="password"          
        />
        <input
          className = "submit-button"
          type = "submit"
        />
      </form>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  handleSignup:(user) => dispatch(updateStoreUser(user)),
  handleError: (error) => dispatch(addError(error))    
});

CreateNewUser.propTypes = {
  handleSignup: PropTypes.func,
<<<<<<< HEAD
  user: PropTypes.object
};
=======
  user: PropTypes.object,
  error: PropTypes.string,
  handleError: PropTypes.func
}
>>>>>>> Add proptypes and fixed api test

export default connect(null, mapDispatchToProps)(CreateNewUser);