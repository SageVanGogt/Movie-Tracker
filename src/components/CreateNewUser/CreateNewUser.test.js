import CreateNewUser from "./CreateNewUser"
import React from 'react'
import {shallow} from 'enzyme'

describe('CreateNewUser', () => {
  let wrapper;
  beforeEach(() => {
    
    wrapper = shallow(<CreateNewUser />)
  });


    it('has a default state', () => {
      let expected = {
        name: "",
        email: "",
        password: ""
      }

      expect(wrapper.state()).toEqual(expected)
    })

    it('should match the snapshot', () => {

      expect(wrapper).toMatchSnapshot()
    })

    it('should update state on change', () => {
      let mockEvent = {
        target: {name: "email", value: "great"}
      }

      let expected = {
        name: "",
        email: "great",
        password: ""
      }
      wrapper.instance().handleChange(mockEvent)

      expect(wrapper.state()).toEqual(expected)

    })

});