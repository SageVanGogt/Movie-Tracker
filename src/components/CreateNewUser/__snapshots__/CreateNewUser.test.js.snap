// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewUser should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateNewUser />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="name"
          onChange={[Function]}
          type="text"
          value=""
/>,
        <input
          name="email"
          onChange={[Function]}
          type="text"
          value=""
/>,
        <input
          name="password"
          onChange={[Function]}
          type="text"
          value=""
/>,
      ],
      "onChange": [Function],
      "type": "submit",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "name",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "email",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "password",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="name"
            onChange={[Function]}
            type="text"
            value=""
/>,
          <input
            name="email"
            onChange={[Function]}
            type="text"
            value=""
/>,
          <input
            name="password"
            onChange={[Function]}
            type="text"
            value=""
/>,
        ],
        "onChange": [Function],
        "type": "submit",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
